google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java

```diff
@@ -1950,7 +1953,17 @@ <ReqT, RespT> GrpcCallContext newCallContext(
       boolean routeToLeader) {
     GrpcCallContext context = GrpcCallContext.createDefault();
     if (options != null) {
+      // Set channel affinity in GAX. This is a no-op for gRPC-GCP.
       context = context.withChannelAffinity(Option.CHANNEL_HINT.getLong(options).intValue());
+
+      // Set channel affinity in gRPC-GCP. This is a no-op for GAX.
```

Comment:
nit: is it possible to only do this if grpc-gcp is being used? I know that it is a no-op for gax, and so does not affect anything in that way, but `context.withCallOptions(..)` always creates a new `CallContext` instance. That means that we reserve a small bit of additional memory for every RPC that we invoke that then needs to be cleaned up again afterwards. (And yes, that's a bit of a pre-emptive optimization, so if it is not possible or hard, then please ignore this comment, but if there's a simple if-condition that we can use, then let's do that.)

Reply:
I have made changes to do it only when grpcgcp is enabled. Can you please take a look?

Reply:
LGTM